`proxy_url` for Elasticsearch output (and possible others) is parsed wrong when scheme is not provided.

- Version: 7.15, 7.16 and master
- Operating System: Linux

## Description
When setting the `proxy_url` for elasticsearch output, and probably others, if no scheme is provided the URL is incorrectly parsed.
The [current documentation states](https://www.elastic.co/guide/en/beats/metricbeat/current/elasticsearch-output.html#_proxy_url) for `proxy_url` states:
> (...) The value may be either a complete URL or a "host[:port]", in which case the "http" scheme is assumed.

However the format `host[:port]` is not supported by the current implementation because it passes the `proxy_url` value directly to `url.Parse`, the [documentation for `url.Parse`](https://pkg.go.dev/net/url#Parse) states:
>Parse parses a raw url into a URL structure.
>
>The url may be relative (a path, without a host) or absolute (starting with a scheme). Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities.

Below is a small snippet that shows the issue ([Go Playground link](https://go.dev/play/p/6ibi2GPXFWE)), at a first glance the parsed URL works, but if one tries to access fields like `Host`, `Path`, `Scheme`, etc, they will notice they're wrongly populated.
```go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u, _ := url.Parse("elastic.co")
	fmt.Printf("Parsed value: %#v\nURL converted to string: '%s'\n\n", u, u.String())

	u, _ = url.Parse("elastic.co:8000")
	fmt.Printf("Parsed value: %#v\nURL converted to string: '%s'\n\n", u, u.String())

	u, _ = url.Parse("elastic.co:8000/path")
	fmt.Printf("Parsed value: %#v\nURL converted to string: '%s'\n\n", u, u.String())

	u, _ = url.Parse("elastic.co/path")
	fmt.Printf("Parsed value: %#v\nURL converted to string: '%s'\n\n", u, u.String())
}
```

## Steps to Reproduce
1. Set a `proxy_url` for an output:
```yaml
output.elasticsearch:
  hosts: ["localhost:9200"]
  proxy_url: "localhost:8888"
````

2. Run the Beat (Metricbeat on this example), you will get some errors. Pay attention to the "DNS lookup failure" log line:
```
2021-12-02T15:58:33.092+0100    INFO    instance/beat.go:309    Setup Beat: metricbeat; Version: 7.15.2
2021-12-02T15:58:33.092+0100    INFO    [index-management]      idxmgmt/std.go:184      Set output.elasticsearch.index to 'metricbeat-7.15.2' as ILM is enabled.
2021-12-02T15:58:33.092+0100    INFO    [elasticsearch] elasticsearch/elasticsearch.go:68       Using proxy URL: localhost:8888
2021-12-02T15:58:33.092+0100    INFO    [esclientleg]   eslegclient/connection.go:100   elasticsearch url: http://localhost:9200
2021-12-02T15:58:33.092+0100    INFO    [publisher]     pipeline/module.go:113  Beat name: x-wing
2021-12-02T15:58:33.092+0100    INFO    eslegclient/connection.go:170   using proxy URL: localhost:8888
2021-12-02T15:58:33.092+0100    INFO    [esclientleg]   eslegclient/connection.go:100   elasticsearch url: http://localhost:9200
2021-12-02T15:58:33.092+0100    WARN    [transport]     transport/tcp.go:52     DNS lookup failure "": lookup : no such host
2021-12-02T15:58:33.092+0100    ERROR   [esclientleg]   transport/logging.go:37 Error dialing lookup : no such host     {"network": "tcp", "address": ":"}
2021-12-02T15:58:33.092+0100    ERROR   [esclientleg]   eslegclient/connection.go:220   error connecting to Elasticsearch at http://localhost:9200: Get "http://localhost:9200": proxyconnect tcp: lookup : no such host                                                                               
2021-12-02T15:58:33.092+0100    ERROR   instance/beat.go:989    Exiting: couldn't connect to any of the configured Elasticsearch hosts. Errors: [error connecting to Elasticsearch at http://localhost:9200: Get "http://localhost:9200": proxyconnect tcp: lookup : no such host]
Exiting: couldn't connect to any of the configured Elasticsearch hosts. Errors: [error connecting to Elasticsearch at http://localhost:9200: Get "http://localhost:9200": proxyconnect tcp: lookup : no such host]
exit status 1
```

--- a/libbeat/common/transport/proxy.go
+++ b/libbeat/common/transport/proxy.go
@@ -67,6 +67,7 @@ func ProxyDialer(log *logp.Logger, config *ProxyConfig, forward Dialer) (Dialer,
 		return nil, err
 	}
 
+	log.Debugf("breaking down proxy URL. Scheme: '%s', host[:port]: '%s', path: '%s'", url.Scheme, url.Host, url.Path)
 	log.Infof("proxy host: '%s'", url.Host)
 	return DialerFunc(func(network, address string) (net.Conn, error) {
 		var err error

--- a/libbeat/esleg/eslegclient/connection.go
+++ b/libbeat/esleg/eslegclient/connection.go
@@ -177,6 +177,7 @@ func NewClients(cfg *common.Config, beatname string) ([]Connection, error) {
 	}
 
 	if proxyURL := config.Transport.Proxy.URL; proxyURL != nil {
+		logp.Debug("breaking down proxy URL. Scheme: '%s', host[:port]: '%s', path: '%s'", proxyURL.Scheme, proxyURL.Host, proxyURL.Path)
 		logp.Info("using proxy URL: %s", proxyURL.URI().String())
 	}
 

--- a/libbeat/outputs/elasticsearch/docs/elasticsearch.asciidoc
+++ b/libbeat/outputs/elasticsearch/docs/elasticsearch.asciidoc
@@ -183,8 +183,7 @@ variables are ignored.
 ===== `proxy_url`
 
 The URL of the proxy to use when connecting to the Elasticsearch servers. The
-value may be either a complete URL or a "host[:port]", in which case the "http"
-scheme is assumed. If a value is not specified through the configuration file
+value must be a complete URL. If a value is not specified through the configuration file
 then proxy environment variables are used. See the
 https://golang.org/pkg/net/http/#ProxyFromEnvironment[Go documentation]
 for more information about the environment variables.

--- a/libbeat/outputs/elasticsearch/elasticsearch.go
+++ b/libbeat/outputs/elasticsearch/elasticsearch.go
@@ -65,6 +65,7 @@ func makeES(
 	}
 
 	if proxyURL := config.Transport.Proxy.URL; proxyURL != nil && !config.Transport.Proxy.Disable {
+		log.Debugf("breaking down proxy URL. Scheme: '%s', host[:port]: '%s', path: '%s'", proxyURL.Scheme, proxyURL.Host, proxyURL.Path)
 		log.Infof("Using proxy URL: %s", proxyURL)
 	}
 

